slice-mapper/src/main/java/com/cognifide/slice/mapper/MapperBuilder.java

```diff
@@ -66,6 +60,12 @@
 	private ChildrenFieldProcessor childrenFieldProcessor;
 
 	@Inject
+	private RequestAttributeProcessor requestAttributeProcessor;
```

Comment:
I'm wondering if this is a good place for this kind of logic. Conceptually, your class needs to be annotated with ```@SliceResource``` only if you want to map its fields from Sling resource. I can imagine a scenario when you don't need mapping to happen but still want to have request parameters/attributes injected. The mapper should not mix these things and fieldProcessors should rely on resource.

If not mapper, then what? I'd probably made RequestAttributeProcessor (and the other one) as Guice providers rather than FieldProcessor. This would make the design cleaner. The disadvantage would be that you'd need to use ```@Inject``` annotation whenever you want the values to be injected and probably specify the name of an attribute (I'm not sure if Guice will give you information about the field name - probably no):
```
@Inject
@RequestAttribute("attribute")
private Boolean attribute;
```
Could you try with this approach?